api/v1/application/repository_latest.go

```diff
@@ -97,7 +98,7 @@ func getWhereClause(criteria map[string]interface{}) string {
 			continue
 		}
 		// TODO SQL injection
-		if key == "search" {
+		if key == "freesearch" {
```

Comment:
This whole algorithm could be genenalized in `hateoas/utils.go` or something in order to be reusable in other repositories.

Reply:
I agree. However, I do not see an easy way to do it right now without a big refactoring. What about doing it outside of this PR ??